### PR TITLE
Add geometric distribution & additional fixes

### PR DESCRIPTION
**Pull request by jmh530**

### CODE REVIEW

**Comment by 9il** [comment]
why the diff is so huge?

**Comment by jmh530** [comment]
Might be easier to look at it on a commit by commit basis. Some of them are
pretty trivial and the diff just looks big because I'm reordering how some
things appear. For instance, it is easier for me to see what is missing
when they are in alphabetical order.

**Comment by 9il** [comment]
Take the index.d for example. The diff is the whole file.
https://github.com/libmir/mir-stat/pull/58/files#diff-ec1aab9dc4cea66c5e2b284f0aff80d1475b6f534411b46015e6a1e435789198

**Comment by jmh530** [comment]
@9il The issue may be related to Windows vs. Unix line endings. I will try to fix when I have time, but it might be a few days.

**Comment by jmh530** [comment]
@9il Fixed.

File: source/mir/stat/distribution/geometric.d [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+See_also: $(LINK2 https://en.wikipedia.org/wiki/Geometric_distribution, geometric probability distribution)
++/
+@safe pure @nogc nothrow
+T geometricPMF(T)(const size_t k, const T p)
+    if (isFloatingPoint!T)
+    in (p >= 0, "p must be greater than or equal to 0")
+    in (p <= 1, "p must be less than or equal to 1")
+{
+    import mir.math.common: pow;
+ 	return pow(1 - p, k) * p;
```
Comment (9il) [inline source/mir/stat/distribution/geometric.d]: Tabs in a few places in the files

Comment (jmh530) [reply source/mir/stat/distribution/geometric.d]: This is what I get for prototyping with `run.dlang.org`

**Pull request merged**